### ISSUE TITLE
bug: unable to build image based on rawhide

### ISSUE THREAD

**Issue by prydom** [blue-build/cli#231]
I've recently been experimenting with updated versions of bluebuild (I'm still on 0.8.4) and am getting the following error:

```
[21:40:28 DEBUG] => Successfully inspected image docker://quay.io/fedora/fedora-kinoite:rawhide!
[21:40:28 ERROR] => Failed:
  × Unable to get the OS version from the labels
  help: Please check with the image author about using
        'org.opencontainers.image.version' to  
```

`org.opencontainers.image.version` label for rawhide is something like "Rawhide.20240927.n.0" which I expect that lenient_semver doesn't support.

I will likely revert https://github.com/blue-build/cli/commit/2bf7c99aaf9ac2fcdb3992a50146bea7a8131fb6 on a personal fork to get myself back up and running.

**Comment by gmpinder** [comment]
Guess we'll have to find a way to set the `OS_VERSION` from inside the container build instead of using inspection. I think the the `lenient_semver` process will only be usable for akmods images for now until we can come up with a more reliable method for retrieving the `OS_VERSION` before creating the Containerfile for building.

**Comment by prydom** [comment]
@gmpinder, Note that it's a slight annoyance that the container tags are tied to the VERSION_ID which may overlap between a Fedora release post cut and rawhide (always rolling) for short time windows, but that is something I can workaround in my CI scripts if needs be. I still consider this closed/completed.

I'm also running into an issue where based on build timings, I think the base image is getting pulled twice. I'll look into that and will file another issue if that seems problematic.

For the record the contents of `/usr/lib/os-release` on Fedora is currently:
```
NAME="Fedora Linux"
VERSION="Rawhide.20240930.n.0 (Kinoite Prerelease)"
ID=fedora
VERSION_ID=42
VERSION_CODENAME=""
PLATFORM_ID="platform:f42"
PRETTY_NAME="Fedora Linux Rawhide.20240930.n.0 (Kinoite Prerelease)"
ANSI_COLOR="0;38;2;60;110;180"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:42"
DEFAULT_HOSTNAME="fedora"
HOME_URL="https://kinoite.fedoraproject.org"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora-kinoite/"
SUPPORT_URL="https://ask.fedoraproject.org/"
BUG_REPORT_URL="https://pagure.io/fedora-kde/SIG/issues"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=rawhide
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=rawhide
SUPPORT_END=2025-05-13
VARIANT="Kinoite"
VARIANT_ID=kinoite
OSTREE_VERSION='Rawhide.20240930.n.0'
```

**Comment by gmpinder** [comment]
> I'm also running into an issue where based on build timings, I think the base image is getting pulled twice. I'll look into that and will file another issue if that seems problematic.

This is a consequence of not being able to get the version of the OS from the labels of the image. In this case I fallback to pulling and running the image to get it. The image pulls and runs outside of the bulidx builder. In my local testing, the buildx builder seemed able to bring the image over from the local registry and just end up extracting it. It's possible however that I didn't properly clear my cache before attempting this. Either way, it's a tradeoff that had to be made. If you can figure out another way to handle it, I'm all ears.

**Comment by gmpinder** [comment]
Just checked again, yeah it's going to end up downloading again when using docker as the build driver.

**Comment by prydom** [comment]
Yep, buildkit and docker store their data in two different places. There's not much we can do about that.

Likewise, unless we add a CLI option to allow specifying the version of the OS being built (or just full customization of the tags in general) I don't see a way around downloading to inspect the image.

I don't seem to get much value, build time wise, from using the GitHub build cache so I was already considering using podman for both runner and builder which I do know defaults to sharing the same storage location. I've not tried that yet with Bluebuild however (I have a separate flow which uses podman build or buildah directly) and I'll report back if it doesn't work for whatever reason.

**Comment by prydom** [comment]
I did confirm that the `podman` builder will share storage with the runner at https://github.com/prydom/my-ostree-build/actions/runs/11166346985/job/31040135532.

I need to work through some incompatibilities, but that looks like my path forward here.

**Comment by prydom** [comment]
It seems that Podman 4.9.3 as is included in Ubuntu 24.04 LTS (current Github Actions runner) is not compatible out-of-the-box with the bluebuild generated containerfile. I've tried to do a `podman system reset -f` in case it was an issue with how intermediate containers were stored.

I am able to successfully use podman as the builder via bluebuild, if I run bluebuild from inside a Fedora container with a more recent version of Podman (5.2.3).

See https://github.com/prydom/my-ostree-build/actions/runs/11168292270, although inspection fails after build. I'll file a bug for that and switch to skopeo for now.

**Comment by gmpinder** [comment]
> It seems that Podman 4.9.3 as is included in Ubuntu 24.04 LTS (current Github Actions runner) is not compatible out-of-the-box with the bluebuild generated containerfile. I've tried to do a `podman system reset -f` in case it was an issue with how intermediate containers were stored.
> 
> I am able to successfully use podman as the builder via bluebuild, if I run bluebuild from inside a Fedora container with a more recent version of Podman (5.2.3).
> 
> See https://github.com/prydom/my-ostree-build/actions/runs/11168292270, although inspection fails after build. I'll file a bug for that and switch to skopeo for now.

This is no longer relevant as shown in #235
